src/modules/item/sagas.ts

```diff
@@ -153,16 +156,43 @@ export function* itemSaga(legacyBuilder: LegacyBuilderAPI, builder: BuilderClien
     }
   }
 
+  function* fetchCollectionItemsWithBatch(collectionId: string, pagesToFetch: number[], limit?: number, isReviewing = false) {
+    const REQUEST_BATCH_SIZE = 10
+    const queue = new PQueue({ concurrency: REQUEST_BATCH_SIZE })
+    const promisesOfPagesToFetch: (() => Promise<PaginatedResource<Item>>)[] = []
+    pagesToFetch.forEach(page => {
+      promisesOfPagesToFetch.push(() => legacyBuilder.fetchCollectionItems(collectionId, page, limit, isReviewing))
+    })
+    const allItemPages: PaginatedResource<Item>[] = yield queue.addAll(promisesOfPagesToFetch)
+    const paginationData = allItemPages[0]
+    const items = allItemPages.map(result => result.results).flat()
+    return { items, paginationData }
+  }
+
   function* handleFetchCollectionItemsRequest(action: FetchCollectionItemsRequestAction) {
-    const { collectionId } = action.payload
+    const DEFAULT_PAGE = 1
+    const { collectionId, page = DEFAULT_PAGE, limit } = action.payload
+    const location: RouterLocation<any> = yield select(getLocation)
+    const isReviewing = location.query.reviewing === 'true'
```

Comment:
we could use the isReviewing selector from `location/selectors.ts` here right?

Reply:
oh, I didn't see that one before, nice!